checkov/kubernetes/runner.py

```diff
@@ -4,49 +4,58 @@
-from checkov.kubernetes.parser.parser import parse
+from checkov.common.runners.base_runner import BaseRunner
+from checkov.kubernetes.graph_builder.local_graph import KubernetesLocalGraph
+from checkov.kubernetes.graph_manager import KubernetesGraphManager
+from checkov.kubernetes.kebernetes_utils import get_files_definitions, get_folder_definitions
 from checkov.kubernetes.checks.resource.registry import registry
 from checkov.runner_filter import RunnerFilter
 
-K8_POSSIBLE_ENDINGS = [".yaml", ".yml", ".json"]
-
 
 class Runner(BaseRunner):
-    check_type = "kubernetes"
+    def __init__(
+        self,
+        graph_class=KubernetesLocalGraph,
+        db_connector=NetworkxConnector(),
+        source="Kubernetes",
+        graph_manager=None,
+    ):
+        self.check_type = "kubernetes"
+        self.graph_class = graph_class
+        # TODO: uncomment it in order to create the graph
+        # self.graph_manager = \
+        #     graph_manager if graph_manager else KubernetesGraphManager(source=source, db_connector=db_connector)
+
+        self.graph_registry = get_graph_checks_registry(self.check_type)
+        self.definitions_raw = {}
 
     def run(self, root_folder, external_checks_dir=None, files=None, runner_filter=RunnerFilter(), collect_skip_comments=True, helmChart=None):
         report = Report(self.check_type)
-        definitions = {}
-        definitions_raw = {}
-        files_list = []
         if external_checks_dir:
             for directory in external_checks_dir:
                 registry.load_external_checks(directory)
+                self.graph_registry.load_external_checks(directory)
 
         if files:
-            _parse_files(files, definitions, definitions_raw)
-
-        if root_folder:
-            filepath_fn = lambda f: f'/{os.path.relpath(f, os.path.commonprefix((root_folder, f)))}'
-            for root, d_names, f_names in os.walk(root_folder):
-                filter_ignored_paths(root, d_names, runner_filter.excluded_paths)
-                filter_ignored_paths(root, f_names, runner_filter.excluded_paths)
-
-                for file in f_names:
-                    file_ending = os.path.splitext(file)[1]
-                    if file_ending in K8_POSSIBLE_ENDINGS:
-                        full_path = os.path.join(root, file)
-                        if "/." not in full_path and file not in ['package.json','package-lock.json']:
-                            # skip temp directories
-                            files_list.append(full_path)
-
-            _parse_files(files_list, definitions, definitions_raw, filepath_fn)
+            definitions, self.definitions_raw = get_files_definitions(files)
+        elif root_folder:
+            definitions, self.definitions_raw = get_folder_definitions(root_folder, runner_filter.excluded_paths)
+        else:
+            return report
+
+        # TODO: uncomment it in order to build the graph
```

Comment:
Why is this commented out?

Reply:
I'll do it in the next task, and already implemented some staff so I don't want to delete